src/coreclr/scripts/genEventPipe.py

```diff
@@ -198,9 +178,12 @@ def generateClrEventPipeWriteEventsImpl(
     WriteEventImpl.append(
         "    EventPipeProvider" +
         providerPrettyName +
-        " = " + createProviderFunc + "(SL(" +
-        providerPrettyName +
-        "Name), " + eventPipeCallbackCastExpr + "(" + callbackName + "));\n")
+        " = " + createProviderFunc + "(")
+    if not runtimeFlavor.nativeaot: WriteEventImpl.append("SL(")
```

Comment:
Mono has `getCoreCLRMonoTypeAdaptionDefines` to avoid doing things like this? Would it make sense to do something similar or reuse for native AOT? It looks `ERROR_SUCCESS` and `ERROR_WRITE_FAULT` are already needed for both.